overlord/ifacestate/ifacestate.go

```diff
@@ -176,56 +201,73 @@ func connect(st *state.State, plugSnap, plugName, slotSnap, slotName string, fla
 	connectInterface.Set("slot-static", slotStatic)
 	connectInterface.Set("plug-dynamic", emptyDynamicAttrs)
 	connectInterface.Set("slot-dynamic", emptyDynamicAttrs)
+	if prepareSlotConnection != nil {
+		connectInterface.WaitFor(prepareSlotConnection)
+	} else {
+		if preparePlugConnection != nil {
```

Comment:
What is the point of the else {  } clause here? I would suspect that it would just be okay to have the else part equally unconditional as the if clause. Is the purpose to untangle a dependency loop if the slot prepare waits for plug prepare? If so, can we have a comment stating as much here? In case I'm wrong a comment that describes the purpose of this would be lovely as well.

Reply:
The only reason is to have simple dependency chain here (and make tests simpler thanks to that); there would be no harm in waiting on both tasks. 
FWTW the old code had simple 1-1 wait dependency as well.

Reply:
I recall seeing a pattern (probably in `snapstate.go:doInstall()`?) where an extra variable, `var last *task.Task` or somesuch was assigned to when adding tasks and used for `WaitFor()`, though this sounds like a material for a follow-up.

Reply:
Right, I know what you mean. I'll see how it fits here in a potential followup PR, thanks.